cpp/oneapi/dal/data/backend/homogen_table_impl.hpp

```diff
@@ -23,28 +23,29 @@ namespace oneapi::dal::backend {
 
 class homogen_table_impl {
 public:
+    homogen_table_impl()
+        : row_count_(0)
+    {}
+
     template <typename DataType>
-    homogen_table_impl(std::int64_t N, std::int64_t p, const DataType* data_pointer, data_layout layout)
-        : row_count_(N),
-          column_count_(p),
-          finfo_(feature_info{ make_data_type<DataType>() }),
-          meta_(table_metadata{ p, finfo_, layout }) {
+    homogen_table_impl(std::int64_t N, std::int64_t p, const DataType* data_pointer, homogen_data_layout layout)
+        : meta_(homogen_table_metadata{ make_data_type<DataType>(), layout, p }),
+          row_count_(N) {
         data_.reset_not_owning(reinterpret_cast<const byte_t*>(data_pointer),
                                N * p * sizeof(DataType));
     }
 
     template <typename DataType, typename = std::enable_if_t<!std::is_pointer_v<DataType>>>
-    homogen_table_impl(std::int64_t N, std::int64_t p, DataType value, data_layout layout)
+    homogen_table_impl(std::int64_t N, std::int64_t p, DataType value, homogen_data_layout layout)
         : homogen_table_impl(N, p, fill_data(new DataType[N*p], N*p, value), layout) {}
 
     template <typename DataType>
-    homogen_table_impl(std::int64_t p, const array<DataType>& data, data_layout layout)
-        : row_count_(data.get_size() / p),
-          column_count_(p),
-          finfo_(feature_info{ make_data_type<DataType>() }),
-          meta_(table_metadata{ p, finfo_, layout }) {
+    homogen_table_impl(std::int64_t p, const array<DataType>& data, homogen_data_layout layout)
+        : meta_(homogen_table_metadata{ make_data_type<DataType>(), layout,  p}),
+          row_count_(data.get_size() / p) {
+        const std::int64_t N = row_count_;
 
-        if (row_count_ * column_count_ != data.get_size()) {
+        if (N * p != data.get_size()) {
```

Comment:
Why `N`, `p` instread `row_count`, `column_count`? Is it worth it all to align?

Reply:
I have renamed it if it is so critical for you. In general, it's an implementation and I can use namings that I want. Here `N` and `p` were very short comparing to `row_count` and `column_count`.